docs/en/examples/elementalarea_with_dataobject.md

```diff
@@ -0,0 +1,244 @@
+
+## Creating Model and adding ElementalArea
+
+In this example the class BlogPost will be created, it will extend DataObject class. Also new BlogPost Admin section will be created in Admin panel where CMS user can manage the BlogPosts. 
+
+Let's look at an example:
+
+**app/src/Admins/BlogPostsAdmin.php**
+
+```php
+<?php
+
+namespace App\Admins;
+
+use App\Models\BlogPost;
+use SilverStripe\Admin\ModelAdmin;
+
+class BlogPostsAdmin extends ModelAdmin
+{
+
+    private static string $url_segment = 'blog-posts-admin';
+
+    private static string $menu_title = 'Blog Posts';
+
+    private static string $menu_icon_class = 'font-icon-block-banner';
+
+    private static array $managed_models = [
+        BlogPost::class,
+    ];
+}
+```
+
+`$managed_models` - this static variable connects BlogPostAdmin with BlogPost, which will be created in the following code.
+When you create DataObject you should define connection between your DataObject and ElementalArea by adding `$has_one` relationship.
+In this example BlogPost will have one ElementalArea.
+
+**app/src/Models/BlogPost.php**
+
+```php
+<?php
+
+namespace App\Models;
+
+use App\Admins\BlogPostsAdmin;
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\ORM\DataObject;
+
+class BlogPost extends DataObject
+{
+    private static string $table_name = 'BlogPost';
+
+    private static array $db = [
+        'Title' => 'Varchar(255)',
+        'Content' => 'HTMLText',
+    ];
+
+    private static $has_one = [
+        'ElementalArea' => ElementalArea::class,
+    ];
+
+    // ...
+}
+ ```
+
+If you are using ElementalArea together with DataObject, it is important to define a few following methods in the class.
+First of all, CMSEditLink() method should be defined to generate link for opening editing section of the particular BlogPost. 
+BaseElement uses link, that was created in CMSEditLink() method in BlogPost class, to generate new links for opening editing section of the particular Element.
+
+```php
+// ...
+
+class BlogPost extends DataObject
+{
+    // ...
+
+    public function CMSEditLink()
+    {
+        // In this example we use BlogPostsAdmin class as Controller
+        $admin = BlogPostsAdmin::singleton();
+
+        // Makes link more readable. Instead App\Models\ BlogPost, we get App-Models-BlogPost
+        $sanitisedClassname = str_replace('\\', '-', $this->ClassName);
+
+        // Returns link to editing section with elements
+        return Controller::join_links(
+            $admin->Link($sanitisedClassname),
+            'EditForm/field/',
+            $sanitisedClassname,
+            'item',
+            $this->ID,
+        );
+    }
+
+    // ...
+}
+
+```
+
+Method Link() doesn't exist in DataObject class, so you should define this method in your class. Since PreviewLink() method of BaseElement uses Link() method to generate Preview window in the BlogPost Admin section. Note: making a DataObject previewable requires silverstripe/admin 1.10 or later.
```

Comment:
Sorry, actually think it may be 1.11 - https://github.com/silverstripe/silverstripe-admin/pull/1250/commits/4603e3c35571485ec8c0bb18900985ff1eaeb7c4 isn't in 1.10

```suggestion
Method Link() doesn't exist in DataObject class, so you should define this method in your class. Since PreviewLink() method of BaseElement uses Link() method to generate Preview window in the BlogPost Admin section. Note: making a DataObject previewable requires silverstripe/admin 1.11 or later.
```